app/ethos/views.py

```diff
@@ -0,0 +1,174 @@
+
+            # Instantiate EthOS contract
+            contract = w3.eth.contract(Web3.toChecksumAddress(settings.ETHOS_CONTRACT_ADDRESS), abi=abi)
+
+            tx = contract.functions.transfer(address, 1 * 10**18).buildTransaction({
+                'nonce': w3.eth.getTransactionCount(settings.ETHOS_ACCOUNT_ADDRESS),
+                'gas': 100000,
+                'gasPrice': recommend_min_gas_price_to_confirm_in_time(5) * 10**9
+            })
+
+            signed = w3.eth.account.signTransaction(tx, settings.ETHOS_ACCOUNT_PRIVATE_KEY)
+            transaction_id = w3.eth.sendRawTransaction(signed.rawTransaction).hex()
+
+            previous_hop = Hop.objects.filter(shortcode=ethos).order_by('-id').first()
+
+            Hop.objects.create(
+                shortcode=ethos,
+                ip=get_ip(request),
+                created_on=timezone.now(),
+                txid=transaction_id,
+                web3_address=address,
+                twitter_username=username,
+                twitter_profile_pic=profile_pic,
+                previous_hop=previous_hop
+            )
+
+            message = transaction_id
+
+            tweet_to_twitter()
+        except ShortCode.DoesNotExist:
+            status = 'error'
+            message = _('Bad request')
+        except Exception as e:
+            status = 'error'
+
+            if 'User not found' in str(e):
+                message = _('Please enter a valid Twitter username')
+            else :
+                message = str(e)
+
+        # http response
+        response = {
+            'status': status,
+            'message': message,
+        }
+
+        return JsonResponse(response)
+
+    try:
+        ShortCode.objects.get(shortcode=shortcode)
+
+        params = {
+            'class': 'redeem',
+            'title': _('EthOS Coin'),
+            'coin_status': _('INITIAL')
+        }
+
+        return TemplateResponse(request, 'redeem_ethos.html', params)
+    except ShortCode.DoesNotExist:
+        raise Http404
+
+
+def tweet_to_twitter():
+    """Tweet the EthOS redemption.
+
+    Args:
+
+
+    Returns:
+        bool: Whether or not the twitter notification was sent successfully.
+
+    """
+    twitter_api = get_twitter_api()
+
+    if not twitter_api:
+        return False
+
+    tweet_txt = "Redeemed EthOS Coin successfully: \n\n #EthOS"
+
+    try:
+        twitter_api.PostUpdate(tweet_txt)
+    except Exception as e:
+        print(e)
+        return False
+
+    return True
+
+
+def get_twitter_api():
+    if not settings.ETHOS_TWITTER_CONSUMER_KEY:
+        return False
+
+    twitter_api = twitter.Api(
```

Comment:
We might want to catch on `twitter.errors.TwitterError` around this API instantiation attempt.